api/internal/localizer/locloader_test.go

```diff
@@ -1,7 +1,7 @@
 // Copyright 2022 The Kubernetes Authors.
 // SPDX-License-Identifier: Apache-2.0
 
-package localizer_test
+package localizer //nolint:testpackage
```

Comment:
Why make this change? I took a scan to see if there are some private elements of `localizer` that you need in this file, but didn't find anything (maybe I missed something).

If it's because you don't like having the `lclzr.` prefix on the exported identifiers, you can get around that by using a dot as the import name like
```
import (
  . "sigs.k8s.io/kustomize/api/internal/localizer"
)
```

Not a big deal, but in general I prefer to not override the linter unless the alternative is significantly worse.

Reply:
Made the change to provide my mock implementation of `processKust`: https://github.com/kubernetes-sigs/kustomize/pull/4859/files/a0d43a7dbee912e232102bd4386d63476a34c7f2#diff-25faa786bf35469d391d5eef99ed1323623023442b177c1ed3e07f3e6709dac6R82

Reply:
ah, kustomize has both `testpackage` and `gochecknoglobals` linters enabled, that means the monkey patching (the var `processKustFn` function to let unittest replace mocked functions)  may not be suitable here, since it always relies on the global variables.
 
Another option is to use interface, but I feel like it's a little bit overkill. 

How about 
1. change those _test.go files to `import localizer_test` (no need to skip `testpackage`). Yes, like natasha suggested you can use `import  . "sigs.k8s.io/kustomize/api/internal/localizer"` to avoid the the localizer prefix, and make some private methods public like `dstPrefix``.
2. we can keep the `//nolint:gochecknoglobals` now, but once you have `processKust` implemented, we probably want to remove the global var function processKustFn to fix the gochecknoglobals linter.

Reply:
This should require the minimum changes to this PR with good test coverage and not skip the kustomize linters (eventually).